src/i2c.rs

```diff
@@ -19,16 +19,38 @@ crate::embedded_hal_error!(
     embedded_hal::i2c::ErrorKind
 );
 
+#[cfg(any(esp32, esp32s2))]
```

Comment:
Aren't you losing precision here? 1 000 000 000 is not divisible by 80 000 000?

Reply:
That one was already used before, I only added the XTAL based 40mhz calculation, but you are right, it is 12.5ns so the old calculation might be wrong by a few clocks